cmd/clusterctl/client/config_test.go

```diff
@@ -745,6 +771,118 @@ func Test_clusterctlClient_GetClusterTemplate_onEmptyCluster(t *testing.T) {
 	}
 }
 
+func newFakeClientWithoutCluster(configClient config.Client) *fakeClient {
+	fake := &fakeClient{
+		configClient: configClient,
+		repositories: map[string]repository.Client{},
+	}
+
+	fake.internalClient, _ = newClusterctlClient("fake-config",
```

Comment:
why is this error ignored?

Reply:
newClusterctlClient does not returns errors in case you are passing a config provider like in this case

Reply:
mm even though is fair enough I wouldn't let the invoker to make such assumption, I'd rather handle the error appropriately or even panic if we want to keep the func signature.

Reply:
@fabriziopandini Let's catch the error and return something, or panic?

Reply:
Sure, done!